central/externalbackups/plugins/s3compatible/s3compatible.go

```diff
@@ -0,0 +1,231 @@
+package s3compatible
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"path/filepath"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/feature/s3/manager"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	s3Types "github.com/aws/aws-sdk-go-v2/service/s3/types"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/central/externalbackups/plugins"
+	"github.com/stackrox/rox/central/externalbackups/plugins/types"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/errorhelpers"
+	"github.com/stackrox/rox/pkg/httputil/proxy"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/urlfmt"
+)
+
+const (
+	backupMaxTimeout               = 4 * time.Hour
+	testMaxTimeout                 = 5 * time.Second
+	initialConfigurationMaxTimeout = 5 * time.Minute
+)
+
+var (
+	log = logging.LoggerForModule()
+)
+
+func init() {
+	plugins.Add(types.S3CompatibleType, func(backup *storage.ExternalBackup) (types.ExternalBackup, error) {
+		return newS3Compatible(backup)
+	})
+}
+
+type s3Compatible struct {
+	integration *storage.ExternalBackup
+
+	client   *s3.Client
+	uploader *manager.Uploader
+}
+
+func validate(cfg *storage.S3Compatible) error {
+	errorList := errorhelpers.NewErrorList("S3 Compatible Validation")
+	if cfg.GetAccessKeyId() == "" {
+		errorList.AddString("Access Key ID must be specified")
+	}
+	if cfg.GetSecretAccessKey() == "" {
+		errorList.AddString("Secret Access Key must be specified")
+	}
+	if cfg.GetRegion() == "" {
+		errorList.AddString("Region must be specified")
+	}
+
+	return errorList.ToError()
+}
+
+func validateEndpoint(endpoint string) error {
+	if _, err := url.Parse(fmt.Sprintf("https://%s", urlfmt.TrimHTTPPrefixes(endpoint))); err != nil {
+		return errors.Wrapf(err, "invalid URL %s", endpoint)
+	}
+	return nil
+}
+
+func newS3Compatible(integration *storage.ExternalBackup) (*s3Compatible, error) {
+	cfg, ok := integration.GetConfig().(*storage.ExternalBackup_S3Compatible)
+	if !ok {
+		return nil, errors.New("S3 Compatible configuration required")
+	}
+	if err := validate(cfg.S3Compatible); err != nil {
+		return nil, err
+	}
+
+	opts := []func(*config.LoadOptions) error{
+		config.WithRegion(cfg.S3Compatible.GetRegion()),
+		config.WithHTTPClient(&http.Client{Transport: proxy.RoundTripper()}),
+		config.WithCredentialsProvider(
+			credentials.NewStaticCredentialsProvider(
+				cfg.S3Compatible.GetAccessKeyId(),
+				cfg.S3Compatible.GetSecretAccessKey(), "",
+			),
+		),
+	}
+
+	ctx, cancelFn := context.WithTimeout(context.Background(), initialConfigurationMaxTimeout)
```

Comment:
Just call it `cancel`:
```suggestion
	ctx, cancel := context.WithTimeout(context.Background(), initialConfigurationMaxTimeout)
```

Reply:
Done